spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplateTests.java

```diff
@@ -134,24 +132,48 @@ public void saveTest() {
 	@Test
 	public void saveAndAllocateIdTest() {
 		TestEntity object = new TestEntity();
-		Entity entity = Entity.newBuilder(createFakeKey()).build();
-		Key key = createFakeKey();
+		Key key = createFakeKey("key");
+		Entity entity = Entity.newBuilder(key).build();
 		when(this.objectToKeyFactory.allocateKeyForObject(same(object), any()))
 				.thenReturn(key);
 		when(this.datastore.put((FullEntity<?>) any())).thenReturn(entity);
-		when(this.datastoreEntityConverter.read(eq(TestEntity.class), same(entity)))
-				.thenReturn(object);
 		assertTrue(this.datastoreTemplate.save(object) instanceof TestEntity);
 		verify(this.datastore, times(1)).put(eq(entity));
 		verify(this.datastoreEntityConverter, times(1)).write(same(object), notNull());
 	}
 
+	@Test
+	public void saveAllAndAllocateIdTest() {
+		TestEntity object1 = new TestEntity();
+		TestEntity object2 = new TestEntity();
+		object2.id = "value";
+
+		Key key1 = createFakeKey("key1");
+		Entity entity1 = Entity.newBuilder(key1).build();
+
+		Key key2 = createFakeKey("key2");
+		Entity entity2 = Entity.newBuilder(key2).build();
+
+		when(this.objectToKeyFactory.allocateKeyForObject(same(object1), any()))
+				.thenReturn(key1);
+		when(this.objectToKeyFactory.getKeyFromObject(same(object2), any()))
+				.thenReturn(key2);
+
+		when(this.datastore.put(any(), any()))
+				.thenReturn(ImmutableList.of(entity1, entity2));
+
+		this.datastoreTemplate.saveAll(ImmutableList.of(object1, object2));
+		verify(this.datastore, times(1)).put(eq(entity1), eq(entity2));
+		verify(this.datastoreEntityConverter, times(1)).write(same(object1), notNull());
+		verify(this.datastoreEntityConverter, times(1)).write(same(object2), notNull());
```

Comment:
should we verify that ids were allocated?

Reply:
That functionality is in the `objectToKeyFactory`, so instead I have a separate unit test class for those that does check that the allocation happens and it is set properly. Here I just am mocking the allocate calls.

Reply:
Oh I see. Maybe rename the test then, because the name suggest that it tests allocate id.